sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs

```diff
@@ -204,13 +209,13 @@ public RehydrationToken GetRehydrationToken()
             AssertNotNull(response, nameof(response));
             AssertNotNull(finalStateVia, nameof(finalStateVia));
 
-            var headerSource = GetHeaderSource(requestMethod, startRequestUri, response, null, out var nextRequestUri);
+            var headerSource = GetHeaderSource(requestMethod, startRequestUri, response, null, out var nextRequestUri, out var isNextRequestPolling);
             string? lastKnownLocation;
             if (!response.Headers.TryGetValue("Location", out lastKnownLocation))
             {
                 lastKnownLocation = null;
             }
-            return GetRehydrationToken(requestMethod, startRequestUri, nextRequestUri, headerSource.ToString(), lastKnownLocation, finalStateVia.ToString(), null);
+            return GetRehydrationToken(requestMethod, startRequestUri, nextRequestUri, headerSource.ToString(), lastKnownLocation, finalStateVia.ToString(), isNextRequestPolling ? ParseOperationId(startRequestUri, nextRequestUri) : null);
```

Comment:
Curious: if there is no id for the operation that is needed to rehydrate, should GetRehydrationToken return null instead of returning a rehydration token with a null id?  How would a rehydration token with a null id be useful to the caller who received it from the call to GetRehydrationToken?

It would be helpful to link to a sample showing how this is used, if one is available.

Reply:
added

Reply:
For instance, a PUT LRO completes in the first request and the next request Uri will be to get the created resource.
In this case, the next request is not polling and the operation id is false.
But if the user doesn't check the status of the initial response, and rehydrate the LRO and try to update the status.
We still enable the user to get the created resource by returning the rehydration token.